apps/src/componentLibrary/checkboxDropdown/CheckboxDropdown.tsx

```diff
@@ -0,0 +1,193 @@
+  useMemo,
+  useRef,
+  useEffect,
+  KeyboardEvent,
+} from 'react';
+import classNames from 'classnames';
+
+import {ComponentSizeXSToL} from '@cdo/apps/componentLibrary/common/types';
+import moduleStyles from './checkboxDropdown.module.scss';
+import Button from '@cdo/apps/templates/Button';
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import i18n from '@cdo/locale';
+
+import Checkbox from '@cdo/apps/componentLibrary/checkbox';
+import FontAwesomeV6Icon from '@cdo/apps/componentLibrary/fontAwesomeV6Icon';
+import {
+  DropdownProvider,
+  useDropdownContext,
+} from '@cdo/apps/componentLibrary/common/contexts/DropdownContext';
+
+export interface CheckboxDropdownProps {
+  /** CheckboxDropdown name */
+  name: string;
+  /** CheckboxDropdown color */
+  color?: 'white' | 'black';
+  /** CheckboxDropdown size */
+  size: ComponentSizeXSToL;
+  /** CheckboxDropdown disabled state */
+  disabled?: boolean;
+  /** CheckboxDropdown label */
+  labelText: string;
+  /** CheckboxDropdown options */
+  allOptions: {value: string; label: string; isOptionDisabled?: boolean}[];
+  /** CheckboxDropdown checked options */
+  checkedOptions: string[];
+  /** CheckboxDropdown onChange handler */
+  onChange: (args: React.ChangeEvent<HTMLInputElement>) => void;
+  /** CheckboxDropdown onSelectAll handler */
+  onSelectAll: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  /** CheckboxDropdown onClearAll handler */
+  onClearAll: (event: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+/**
+ * ### Production-ready Checklist:
+ * * (✔) implementation of component approved by design team;
+ * * (✔) has storybook, covered with stories and documentation;
+ * * (✔) has tests: test every prop, every state and every interaction that's js related;
+ * * (see apps/test/unit/componentLibrary/DropdownMenuTest.jsx)
+ * * (?) passes accessibility checks;
+ *
+ * ###  Status: ```Ready for dev```
+ *
+ * Design System: Checkbox Dropdown Component.
+ * Used to render checkbox (multiple choice) dropdowns.
+ */
+const CheckboxDropdown: React.FunctionComponent<CheckboxDropdownProps> = ({
+  name,
+  labelText,
+  allOptions,
+  checkedOptions = [],
+  onChange,
+  onSelectAll,
+  onClearAll,
+  disabled = false,
+  color = 'black',
+  size = 'm',
+}) => {
+  const {activeDropdownName, setActiveDropdownName} = useDropdownContext();
+  const dropdownRef: React.MutableRefObject<HTMLDivElement | null> =
+    useRef(null);
+
+  const handleClickOutside = useCallback(
+    (event: Event) => {
+      if (
+        activeDropdownName &&
+        dropdownRef.current &&
+        event.target instanceof Node &&
+        !dropdownRef.current.contains(event.target)
+      ) {
+        setActiveDropdownName('');
+      }
+    },
+    [dropdownRef, setActiveDropdownName, activeDropdownName]
+  );
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+    document.addEventListener('keydown', handleClickOutside);
```

Comment:
Does this close the dropdown for any key press?

Reply:
yes, for any key press. Works as expected since it checks the event target attribute

Reply:
Hi Denys, I pulled your branch and 'enter' and 'space bar' seemed like the only keys that to close an open dropdown.

Reply:
Not sure if this is within the scope of this PR, but currently when a dropdown option is highlighted, the 'Enter' key press does check the checkbox.

Reply:
Hey @fisher-alice , for open/close dropdown - that's correct behavior. The part you comment refers only the logic behind closeing the dropdown when clicked/keydowned anywhere outside of the dropdown content (can be any button which is also ok)

As for checkbox - that's also an intended behaviour since that's default browser approach for input fields (which checkboxes are)